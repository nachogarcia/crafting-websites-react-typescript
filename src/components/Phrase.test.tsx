import * as React from 'react';
import {shallow, ShallowWrapper} from "enzyme";
import PhraseComponent from './Phrase'

const text = 'some joke'

describe('Phrase', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<PhraseComponent text={text}/>)
    });

    it('displays joke', () => {
        expect(wrapper.contains(text)).toBe(true);
    });

    it('has correct structure', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});