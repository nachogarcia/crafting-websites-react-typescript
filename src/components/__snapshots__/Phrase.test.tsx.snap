// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phrase has correct structure 1`] = `

<div class="phrase">
  <span>
    some joke
  </span>
  <hr>
</div>

`;
